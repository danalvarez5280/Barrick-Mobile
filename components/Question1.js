import React, { Component } from 'react';
import logo from './assets/caution.png';
import truck from './assets/truck.png';
import truckGold from './assets/truck-gold.png';
import wrench from './assets/wrench.png';
import wrenchGold  from './assets/wrench-gold.png';
import light from './assets/light.png';
import lightGold from './assets/light-gold.png';

import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

//colors
// gold #C4900F
// blue #122732
//off white #CED5E0
// off blue #324A5E

import {
  Alert,
  AppRegistry,
  Button,
  Image,
  Keyboard,
  Slider,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';

class Question1 extends Component {
  constructor() {
    super();
    this.state = {
      taskType: '',
      specificTask: ''
    }
    this.handleSubmit = this.handleSubmit.bind(this);
    this.noInfo = this.noInfo.bind(this);
    this.selectTaskType = this.selectTaskType.bind(this);
  }

  handleSubmit() {
    console.log('hi everybody');
    this.props.submitLogin(this.state)
  };

  noInfo() {
    Alert.alert('You need to input an TASK TYPE  and SPECIFIC TASK')
  };

  selectTaskType(type) {
    this.setState({
      taskType: type,
    })
    console.log(this.state.taskType);
  };

  render() {
    const transportation = this.state.taskType === 'Transportation' ? truckGold : truck;
    const electrical = this.state.taskType === 'Electrical' ? lightGold : light;
    const maintenence = this.state.taskType === 'Maintenence' ? wrenchGold : wrench;
    const submitButton =
      this.state.specificTask.length ?
        <Button style={ styles.yesSubmit } onPress={ this.handleSubmit } title='Submit'/> :
        <Button style={ styles.noSubmit } onPress={ this.noInfo } title='Submit'/>;
    return(
      <KeyboardAwareScrollView
        resetScrollToCoords={{ x: 0, y: 0 }}
        scrollEnabled={ true }
        contentContainerStyle={ styles.display }>
        <View style={ styles.question } >
          <Text style={ styles.prompt }>Select the Type of Task</Text>
          <View style={ styles.taskType }>
            <View>
              <TouchableHighlight onPress={ () => this.selectTaskType('Transportation') }>
                <Image source={ transportation } style={styles.taskIcon}/>
              </TouchableHighlight>
              <Text style={styles.type}>Transportation</Text>
            </View>
            <View>
              <TouchableHighlight onPress={ () => this.selectTaskType('Electrical') }>
                <Image source={ electrical } style={styles.taskIcon}/>
              </TouchableHighlight>
              <Text style={styles.type}>Electrical</Text>
            </View>
            <View>
              <TouchableHighlight onPress={ () => this.selectTaskType('Maintenence') }>
                <Image source={ maintenence } style={styles.taskIcon}/>
              </TouchableHighlight>
              <Text style={styles.type}>Maintenence</Text>
            </View>

          </View>
          <Text style={ styles.prompt }>Specific Task</Text>
          <TextInput
            style={ styles.input }
            multiline= { true }
            placeholder='Specific Task'
            onChangeText={ (employee) => this.setState({ employee })}
            value={ this.state.employee } />
          <Button onPress={ this.handleSubmit } title='Submit'/>
        </View>
      </KeyboardAwareScrollView>
    )
  }
};
const styles = StyleSheet.create({
  display: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop: 10,
    height: '100%',
    width: '100%',
  },
  input: {
    backgroundColor: '#CED5E0',
    color: '#122732',
    fontSize: 20,
    height: 150,
    marginBottom: 10,
    padding: 5,
    width: 250,
  },
  prompt: {
    backgroundColor: '#C4900F',
    color: '#122732',
    fontWeight: 'bold',
    padding: 5,
    textAlign: 'center',
    width: 250,
  },
  question: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
  taskIcon: {
    height: 80,
    margin: 15,
    width: 80,
  },
  taskType: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginBottom: 10,
  },
  type: {
    color: '#122732',
    display: 'flex',
    fontSize: 10,
    fontWeight: 'bold',
    justifyContent: 'center',
    textAlign: 'center',
  }

});

export default Question1;
