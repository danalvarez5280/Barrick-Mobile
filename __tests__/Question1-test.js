import 'react-native';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Question1 from '../components/Question1';
import { Button, Image, Text, TextInput, TouchableHighlight } from 'react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const q1Submit = jest.fn();
const setControls = jest.fn(args => console.log("i got called! " + args));
const setRisks = jest.fn(args => console.log("i got called, sucka! " + args));
const logOut = jest.fn();
// const selectTaskType = jest.fn();
const mockState = {
  specificTask: "Electrical"
};

const setup = () => {
  return shallow(<Question1
    submitLogin={ q1Submit }
    setControls={ setControls }
    setRisks={ setRisks }
    info={ mockState }
    logOut={ logOut } />)
};

const resetMocks = () => {
  setControls.mockClear()
  setRisks.mockClear()
}

beforeEach(() => {
  resetMocks()
})

it('renders correctly', () => {

  const tree = renderer.create(
    <Question1
    submitLogin={ q1Submit }
    setControls={ setControls }
    setRisks={ setRisks }
    info={ mockState }
    logOut={ logOut }/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should have 5 texts displayed', () => {
  let question1 = setup();

  expect(question1.find(Text).length).toEqual(5);
});

it('should have a text input area', () => {
  let question1 = setup();

  expect(question1.find(TextInput).length).toEqual(1);
});

it('should have an 3 images, one for each task type, each one should be a touchable TouchableHighlight', () => {
  let question1 = setup();

  expect(question1.find(Image).length).toEqual(3);
  expect(question1.find(TouchableHighlight).length).toEqual(3);
});

it('should be able to set the task type', () => {
  let question1 = setup();

  const selectTaskTypeSpy = jest.spyOn(question1.instance(), 'selectTaskType').mockImplementation(() => {});
  expect(question1.find(TouchableHighlight).length).toEqual(3);
  question1.find(TouchableHighlight).first().simulate('press');
  expect(selectTaskTypeSpy).toHaveBeenCalledWith('Transportation');
  question1.find(TouchableHighlight).last().simulate('press');
  expect(selectTaskTypeSpy).toHaveBeenCalledWith('Maintenance');
  question1.find(TouchableHighlight).at(1).simulate('press');
  expect(selectTaskTypeSpy).toHaveBeenCalledWith('Electrical');
});


it('can log out the user', () => {
  let question1 = setup();

  expect(question1.find(Button).length).toEqual(2);
  question1.find(Button).last().simulate('press');
  expect(logOut).toHaveBeenCalled();
});

it('should be able to submit the answers to question 1', () => {
  let question1 = setup();

  expect(question1.find(Button).length).toEqual(2);
  question1.find(Button).first().simulate('press');
  expect(q1Submit).toHaveBeenCalled();
});

it('should set the electric risks when they are fetched', async () => {
  fetch.mockResponse(JSON.stringify([{body: 'thing1'}, {body: 'thing2'}]))
  let question1 = setup();

  await question1.instance().getElectricRisks()
  expect(setRisks).toHaveBeenCalledWith(['thing1', 'thing2']);
  expect(setRisks).toHaveBeenCalled();
});

it('sets electric controls when they are fetched', async () => {
  fetch.mockResponse(JSON.stringify([{body: 'thing1'}, {body: 'thing2'}]))
  let question1 = setup();

  await question1.instance().getElectricControls()
  expect(setControls).toHaveBeenCalledWith(['thing1', 'thing2'])
  expect(setControls).toHaveBeenCalled();
});
