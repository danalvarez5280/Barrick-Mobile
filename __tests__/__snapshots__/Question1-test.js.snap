// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": 400,
      "justifyContent": "space-between",
      "marginTop": 0,
      "width": "100%",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
            "width": "80%",
          }
        }
      >
        Select the Type of Task
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginBottom": 10,
          }
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../components/assets/truck.png",
                }
              }
              style={
                Object {
                  "height": 80,
                  "margin": 15,
                  "width": 80,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#122732",
                "display": "flex",
                "fontSize": 10,
                "fontWeight": "bold",
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            Transportation
          </Text>
        </View>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../components/assets/light.png",
                }
              }
              style={
                Object {
                  "height": 80,
                  "margin": 15,
                  "width": 80,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#122732",
                "display": "flex",
                "fontSize": 10,
                "fontWeight": "bold",
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            Electrical
          </Text>
        </View>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../components/assets/wrench.png",
                }
              }
              style={
                Object {
                  "height": 80,
                  "margin": 15,
                  "width": 80,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#122732",
                "display": "flex",
                "fontSize": 10,
                "fontWeight": "bold",
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            Maintenance
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
            "width": "80%",
          }
        }
      >
        Specific Task
      </Text>
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Specific Task"
        style={
          Object {
            "backgroundColor": "#CED5E0",
            "color": "#122732",
            "fontSize": 20,
            "height": 100,
            "marginBottom": 10,
            "padding": 5,
            "width": "80%",
          }
        }
        value="Electrical"
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            SUBMIT
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            LOG OUT
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
